gazebosim/gz-sim
# Script and tutorial for generating procedural datasets with Blender

# 🎉 New feature

## Summary
Hello :smile:. I recently saw https://github.com/ignitionrobotics/ign-gazebo/pull/1401 so I thought that I would share how I have used Blender to generate procedural datasets for one of my projects. The video below should explain the gist of it.

https://user-images.githubusercontent.com/22929099/160182025-e1868b54-f3f1-44ee-9ea6-b62e181e8af5.mp4

For this request, I modified my existing setup into a script that directly exports SDF models (inspired by [Ignition tutorial for exporting SDF exporting from Blender](https://github.com/ignitionrobotics/ign-gazebo/blob/9b07b245e8384628fa20d09a563c48a6a063c0c0/tutorials/blender_sdf_exporter.md)). I included a "[tutorial](https://github.com/AndrejOrsula/ign-gazebo/blob/800eda6f36aa63e347e8d5b03cf49ecb6bff8c9a/tutorials/blender_procedural_datasets.md)" to go alongside it with hope that other contributors would share their `.blend` files in the future. This tutorial contains the important details, so I won't repeat them here.

I am not exactly sure what would be the best way to integrate this "feature" with Ignition Fuel without uploading terabytes of datasets. Maybe it could be discussed below if this request is of any interest.

## Test it
- [Install Blender 3.0 or 3.1](https://www.blender.org/download) (or try your luck with other versions)
- Open [rock.blend](https://github.com/AndrejOrsula/ign-gazebo/blob/800eda6f36aa63e347e8d5b03cf49ecb6bff8c9a/tutorials/files/blender_procedural_generation/rock.blend)
- Execute the embedded Python script
- See the results

Note: `rock.blend` was my first attempt at utilising Geometry Nodes. The procedural pipeline is therefore far from optimal. Maybe using one of the official demos from https://www.blender.org/download/demo-files/#geometry-nodes and modifying it for dataset generation would be better to serve as an example.

## Checklist

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.
